python/lsst/ts/ATWhiteLightSource/wlsCSC.py

```diff
@@ -213,49 +254,47 @@ def get_config_pkg():
                 -------
                 None
             """
+        if self.kiloarcModel.component is not None:
+            if not self.kiloarcModel.component.client.connect():
+                raise salobj.ExpectedError("Can't stop chillin' when we're disconnected from the Kiloarc's ADAM; we don't know if the lamp is still running and in need of cooling.")
         if self.kiloarcModel.bulb_on:
-            raise salobj.ExpectedError("Can't stop chillin' while the bulb is still on.")
+            raise salobj.ExpectedError("Can't stop chillin' while the bulb is still on, or bulb state is unknown.")
+        if not self.keep_on_chillin_task.done():
+            remaining = round(self.config.keep_on_chillin_timer - (time.time() - self.lamp_off_time),0) + 5
+            raise salobj.ExpectedError(f"Lamp was recently extinguished; can't stop chillin' for {remaining} seconds.")
         else:
             await self.chillerModel.stopChillin()
 
-    async def stateloop(self):
-        """
-        periodically prints the current state. For debug
-        """
-        while self.stateLoopBool:
-            print("current state:  "+str(self.summary_state))
-           # print("detailed state: "+str(self.detailed_state))
-           # print(self.kiloarcListenerTask)
-           # print(self.telemetryLoopTask)
-            await asyncio.sleep(1)
-
     async def kiloarc_interlock_loop(self):
         """
         Make sure we stop the bulb if something bad happens with the chiller
         """
         while self.interlockLoopBool:
+            # chiller alarms will take us to FAULT even if bulb is off
+            if self.chillerModel.alarmPresent == AlarmStatus.ALARM:
+                    currentAlarms = self.chillerModel.l1AlarmsPresent + self.chillerModel.l2AlarmsPresent
+                    self.log.debug("Chiller Reporting Alarm: " + str(currentAlarms))
+                    self.fault(report="Chiller Reporting Alarm: " + str(currentAlarms))
+            # if the bulb is on and something goes wrong with chiller, e-stop the bulb. 
             if self.kiloarcModel.bulb_on:
-                if self.chillerModel.alarmPresent:
-                    print("**Alarm Present")
-                    self.summary_state = salobj.State.FAULT
+                if self.chillerModel.alarmPresent == AlarmStatus.ALARM:
+                    self.fault()
                     await self.kiloarcModel.emergencyPowerLightOff()
                 if self.chillerModel.chillerStatus != 1:
-                    self.summary_state = salobj.State.FAULT
-                    print("**Chiller Status not RUN")
+                    self.fault()
```

Comment:
When calling `self.fault` pass `code` and `report` so `errorCode` event is published with information about the `FAULT`

Reply:
tenatively going to lump all chiller related alarms under code 2, rather than giving every possible fault its own error code...